ic-os/guestos/defs.bzl

```diff
@@ -28,29 +28,29 @@ def image_deps(mode, malicious = False):
-            "//publish/binaries:fstrim_tool": "/opt/ic/bin/fstrim_tool:0755",
-            "//publish/binaries:guestos_tool": "/opt/ic/bin/guestos_tool:0755",
-            "//publish/binaries:ic-btc-adapter": "/opt/ic/bin/ic-btc-adapter:0755",
-            "//publish/binaries:ic-consensus-pool-util": "/opt/ic/bin/ic-consensus-pool-util:0755",
-            "//rs/https_outcalls/adapter:ic-https-outcalls-adapter": "/opt/ic/bin/ic-https-outcalls-adapter:0755",  # `//publish/binaries:ic-https-outcalls-adapter` is for testing and must NOT be used here
-            "//publish/binaries:ic-crypto-csp": "/opt/ic/bin/ic-crypto-csp:0755",
-            "//publish/binaries:ic-regedit": "/opt/ic/bin/ic-regedit:0755",
-            "//publish/binaries:ic-recovery": "/opt/ic/bin/ic-recovery:0755",
-            "//publish/binaries:orchestrator": "/opt/ic/bin/orchestrator:0755",
-            "//publish/binaries:ic-boundary-tls": "/opt/ic/bin/ic-boundary:0755",
-            ("//publish/malicious:replica" if malicious else "//publish/binaries:replica"): "/opt/ic/bin/replica:0755",  # Install the malicious replica if set
-            "//publish/binaries:metrics-proxy": "/opt/ic/bin/metrics-proxy:0755",
-            "//publish/binaries:nft-exporter": "/opt/ic/bin/nft-exporter:0755",
-            "//publish/binaries:sandbox_launcher": "/opt/ic/bin/sandbox_launcher:0755",
-            "//publish/binaries:state-tool": "/opt/ic/bin/state-tool:0755",
-            "//publish/binaries:vsock_guest": "/opt/ic/bin/vsock_guest:0755",
-            "//cpp:infogetty": "/opt/ic/bin/infogetty:0755",
-            "//cpp:prestorecon": "/opt/ic/bin/prestorecon:0755",
+            "//publish/binaries:canister_sandbox": "/opt/ic/bin/canister_sandbox:0755",  # Need for the canister sandboxing to work.
+            "//publish/binaries:compiler_sandbox": "/opt/ic/bin/compiler_sandbox:0755",  # Need for the Wasm compilation sandboxing to work.
+            "//publish/binaries:sandbox_launcher": "/opt/ic/bin/sandbox_launcher:0755",  # Need for the canister/compilation sandboxing to work.
+            "//publish/binaries:ic-btc-adapter": "/opt/ic/bin/ic-btc-adapter:0755",  # Need for the Bitcoin integration.
+            "//rs/https_outcalls/adapter:ic-https-outcalls-adapter": "/opt/ic/bin/ic-https-outcalls-adapter:0755",  # Need for the HTTPS outcalls feature. `//publish/binaries:ic-https-outcalls-adapter` is for testing and must NOT be used here
+            "//publish/binaries:ic-crypto-csp": "/opt/ic/bin/ic-crypto-csp:0755",  # Crypto operations provider, required by the IC protocol (signing, etc).
+            "//publish/binaries:orchestrator": "/opt/ic/bin/orchestrator:0755",  # Replica process manager, required by the IC protocol (upgrades, node addition, etc).
+            ("//publish/malicious:replica" if malicious else "//publish/binaries:replica"): "/opt/ic/bin/replica:0755",  # Main protocol binary, required by the IC protocol. Installs the malicious replica if set.
```

Comment:
Could you please add a couple words about how "malicious" replica is only used in test builds?